src/datasets/domain/models/Dataset.ts

```diff
@@ -0,0 +1,40 @@
+export interface Dataset {
+  id: number;
+  persistentId: string;
+  versionId: number;
+  versionInfo: DatasetVersionInfo;
+  license: DatasetLicense;
+  metadataBlocks: DatasetMetadataBlock[];
+}
+
+export interface DatasetVersionInfo {
+  majorNumber: number;
+  minorNumber: number;
+  state: DatasetVersionState;
+  createTime: Date;
+  lastUpdateTime: Date;
+  releaseTime?: Date;
+}
+
+export enum DatasetVersionState {
+  DRAFT = 'DRAFT',
+  RELEASED = 'RELEASED',
+  ARCHIVED = 'ARCHIVED',
+  DEACCESSIONED = 'DEACCESSIONED',
+}
+
+export interface DatasetLicense {
+  name: string;
+  uri: string;
+}
+
+export interface DatasetMetadataBlock {
+  name: string;
+  fields: DatasetMetadataFields;
+}
+
+export type DatasetMetadataFields = Record<string, DatasetMetadataFieldValue>;
+
+export type DatasetMetadataFieldValue = string | string[] | DatasetMetadataSubField | DatasetMetadataSubField[];
```

Comment:
```suggestion
export type DatasetMetadataFieldValue = string[] | DatasetMetadataSubField[];
```

why we do not simplify this type to this? so when there is only 1 element is an array with one value, when there are many elements is an array with more than 1 element, less logic to handle

Reply:
Ok, I spoke with Guillermo and actually the initial approach makes sense, I'll make the changes in the frontend to meet this format

`export type DatasetMetadataFieldValue = string | string[] | DatasetMetadataSubField | DatasetMetadataSubField[];`